google/resource_storage_bucket.go

```diff
@@ -73,6 +75,64 @@ func resourceStorageBucket() *schema.Resource {
 				ForceNew: true,
 			},
 
+			"lifecycle_rule": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"action": {
+							Type:     schema.TypeSet,
+							Required: true,
+							MaxItems: 1,
+							Set:      resourceGCSBucketLifecycleRuleActionHash,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"type": {
+										Type:     schema.TypeString,
+										Required: true,
+									},
+									"storage_class": {
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+								},
+							},
+						},
+						"condition": {
+							Type:     schema.TypeSet,
+							Required: true,
+							MaxItems: 1,
+							Set:      resourceGCSBucketLifecycleRuleConditionHash,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"age": {
+										Type:     schema.TypeInt,
+										Optional: true,
+									},
+									"created_before": {
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+									"is_live": {
+										Type:     schema.TypeBool,
+										Optional: true,
+									},
+									"matches_storage_class": {
+										Type:     schema.TypeList,
+										Optional: true,
+										Elem:     &schema.Schema{Type: schema.TypeString},
+									},
+									"number_of_newer_versions": {
```

Comment:
The api describes this field as 'numNewerVersions' so I suppose it should be 'num_newer_versions' for consistency sake (although I do prefer this name, it's probably better to match the API).